tests/TestCase/ORM/RulesCheckerIntegrationTest.php

```diff
@@ -1348,12 +1349,15 @@ public function testIsLinkedToInferFieldFromAssociationNameWithNoRepositoryAvail
     {
         $rulesChecker = new RulesChecker();
 
-        /** @var \Cake\ORM\Table|\PHPUnit\Framework\MockObject\MockObject $Comments */
-        $Comments = $this->getMockForModel('Comments', ['rulesChecker'], ['className' => Table::class]);
-        $Comments
-            ->expects($this->any())
-            ->method('rulesChecker')
-            ->willReturn($rulesChecker);
+        $Comments = Mockery::mock(
+            Table::class . '[rulesChecker]',
+            [[
+                'alias' => 'Comments',
+                'table' => 'comments',
+            ]]
+        );
```

Comment:
Should we make a shared wrapper for this?

Reply:
Not anymore since I got rid of the mock usage altogether.

It also unearthed a hidden issue. For some reason earlier the field name in the error message was `articles` when belongsTo property names are supposed to be singular. The assertion is now correctly updated to expect `article`. Mocks are evil!